packages/dd-trace/src/config.js

```diff
@@ -772,9 +503,67 @@ ken|consumer_?(?:id|key|secret)|sign(?:ed|ature)?|auth(?:entication|orization)?)
     this._setBoolean(defaults, 'logInjection', false)
     this._setArray(defaults, 'headerTags', [])
     this._setValue(defaults, 'tags', {})
+    this._setBoolean(defaults, 'tracing', true)
+    this._setValue(defaults, 'dbmPropagationMode', 'disabled')
+    this._setBoolean(defaults, 'dsmEnabled', false)
+    this._setBoolean(defaults, 'openAiLogsEnabled', false)
+    this._setValue(defaults, 'url', undefined)
+    this._setValue(defaults, 'site', 'datadoghq.com')
```

Comment:
does `site` follow the same logic? 
```
coalesce(options.site, process.env.DD_SITE, 'datadoghq.com')
```
it seems the options took precedence but now `_applyEnvironment` is called before `_applyOptions`

Reply:
`_applyEnvironment` and `_applyOptions` don't actually add to `this` (i.e. the Config object), `[_merge](https://github.com/DataDog/dd-trace-js/blob/f0b97ef55926306a141675998b1f2a80c80df71d/packages/dd-trace/src/config.js#L876)` does, which applies options before environment when it is not null/undefined

Reply:
thanks for the context 👍